packages/gestalt/src/Typeahead.js

```diff
@@ -0,0 +1,351 @@
+// @flow strict
+import React, { useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import InputField from './TypeaheadInputField.js';
+import Option from './TypeaheadOption.js';
+import Box from './Box.js';
+import Text from './Text.js';
+import Flyout from './Flyout.js';
+import Layer from './Layer.js';
+
+type TextColors =
+  | 'green'
+  | 'pine'
+  | 'olive'
+  | 'blue'
+  | 'navy'
+  | 'midnight'
+  | 'purple'
+  | 'orchid'
+  | 'eggplant'
+  | 'maroon'
+  | 'watermelon'
+  | 'orange'
+  | 'darkGray'
+  | 'gray'
+  | 'lightGray'
+  | 'red'
+  | 'white';
+
+type OptionObject = {
```

Comment:
@simkessy could we make these [exact](https://flow.org/en/docs/types/objects/#toc-exact-object-types)? Just to maintain the same syntax in this file `{| |}`

Reply:
Sure but can I get a better understanding into why we force a certain object type into the component? If a user's data doesn't return in the format of `label/value` why are we forcing them into it and what benefit does that provide the user?